release_dashboard/lib/widgets/engine_cherrypicks_substeps.dart

```diff
@@ -0,0 +1,99 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+
+import 'common/checkbox_substep.dart';
+
+enum EngineCherrypicksSubstep {
+  substep1,
+  substep2,
+  substep3,
+}
+
+/// Group and display all substeps related to the 'Apply Engine Cherrypicks' step into a widget.
+///
+/// When all substeps are completed, [nextStep] can be executed to proceed to the next step.
+class EngineCherrypicksSubsteps extends StatefulWidget {
+  const EngineCherrypicksSubsteps({
+    Key? key,
+    required this.nextStep,
+  }) : super(key: key);
+
+  final VoidCallback nextStep;
+
+  @override
+  State<EngineCherrypicksSubsteps> createState() => ConductorSubstepsState();
+
+  static Map<EngineCherrypicksSubstep, String> substepTitles = <EngineCherrypicksSubstep, String>{
+    EngineCherrypicksSubstep.substep1: 'Substep 1',
+    EngineCherrypicksSubstep.substep2: 'Substep 2',
+    EngineCherrypicksSubstep.substep3: 'Substep 3',
+  };
+
+  static Map<EngineCherrypicksSubstep, String> substepSubtitles = <EngineCherrypicksSubstep, String>{
+    EngineCherrypicksSubstep.substep1: 'Substep subtitle 1',
+    EngineCherrypicksSubstep.substep2: 'Substep subtitle 2',
+    EngineCherrypicksSubstep.substep3: 'Substep subtitle 3',
+  };
+}
+
+class ConductorSubstepsState extends State<EngineCherrypicksSubsteps> {
+  Map<String, bool> _isEachSubstepChecked = <String, bool>{};
```

Comment:
Which key are you talking about? Each `CheckboxAsSubstep` has a key initialized [here](https://github.com/Yugue/cocoon/blob/6cf0fdbe82a5b2eaa70248aab88341bebff6eef6/release_dashboard/lib/widgets/common/checkbox_substep.dart#L27)

Reply:
you re-requested a review, was this comment addressed?

Reply:
oops, I didn't refresh, I see your comment now.

Reply:
> Are you referring to the key in `CheckboxAsSubstep`? Each `CheckboxAsSubstep` has a key initialized [here](https://github.com/Yugue/cocoon/blob/6cf0fdbe82a5b2eaa70248aab88341bebff6eef6/release_dashboard/lib/widgets/common/checkbox_substep.dart#L27). Or is it something else?

I mean the key to the `_isEachSubstepChecked` map.

Reply:
Yes, I am using the enum `EngineCherrypickSubstep` as key to access the `_isEachSubstepChecked` map [here](https://github.com/Yugue/cocoon/blob/6cf0fdbe82a5b2eaa70248aab88341bebff6eef6/release_dashboard/lib/widgets/engine_cherrypicks_substeps.dart#L51) when initializing (before I was directly using the values instead of the key). Also I am using the key to update the map [here](https://github.com/Yugue/cocoon/blob/6cf0fdbe82a5b2eaa70248aab88341bebff6eef6/release_dashboard/lib/widgets/engine_cherrypicks_substeps.dart#L72).

Reply:
If you used `EngineChrerypickSubstep` as the key of `_isEachSubstepChecked` instead of the title, you wouldn't have to dereference it.

Reply:
Oh, I see what you mean. I was not using the enum directly. That's a really good suggestion. Changed!!

Reply:
My bad, didn't realize I had two layers of logic to get the string as the key.